subprojects/core/src/test/groovy/org/gradle/util/NameMatcherTest.java

```diff
@@ -121,7 +121,7 @@ public void doesNotSelectItemsWhenNoMatches() {
         assertDoesNotMatch("name", "other");
         assertDoesNotMatch("name", "na");
         assertDoesNotMatch("sN", "otherName");
-        assertDoesNotMatch("sN", "someThing");
```

Comment:
We use a case insensitive pattern matcher `\Qs\E[\p{javaLowerCase}\p{Digit}]*\QN\E[\p{javaLowerCase}\p{Digit}]*\Q\E
` for "sN" shorthand which should match "**s**omeThi**n**g"